File: Task1/Task1.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function printPattern(N) {
    const ch = String.fromCharCode('A'.charCodeAt(0) + N - 1);
    for (let i = N; i >= 1; --i) {
        let line = '';
        for (let j = 0; j < N - i + 1; ++j) {
            line += String.fromCharCode(ch.charCodeAt(0) - j) + ' ';
        }
        console.log(line);
    }
}

console.log("Введите значение N (положительное число): ");

rl.on('line', (input) => {
    const N = parseInt(input);
    if (N > 0) {
        printPattern(N);
        rl.close();
    } else {
        console.log("Введите корректное положительное значение: ");
    }
});

File: Task1/Task1.kt
fun printPattern(N: Int) {
    val ch = 'A' + (N - 1)
    for (i in N downTo 1) {
        for (j in 0 until (N - i + 1)) {
            print("${ch - j} ")
        }
        println()
    }
}

fun main() {
    print("Введите число N (положительное число): ")
    val N = readLine()?.toIntOrNull()
    if (N != null && N > 0) {
       printPattern(N)
    } else {
        println("Неверный ввод")
    }
}
